Firestore/src/FirestoreSessionHandler.php

```diff
@@ -0,0 +1,404 @@
+ * Pass a positive value up to 1000 for $gcLimit parameter to delete entities in
+ * gc.
+ *
+ * The first example automatically writes the session data. It's handy, but
+ * the code doesn't stop even if it fails to write the session data, because
+ * the `write` happens when the code exits. If you want to know whether the
+ * session data is correctly written to Firestore, you need to call
+ * `session_write_close()` explicitly and then handle `E_USER_WARNING`
+ * properly. See the second example for a demonstration.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Firestore\FirestoreClient;
+ *
+ * $firestore = new FirestoreClient();
+ *
+ * $handler = $firestore->sessionHandler();
+ *
+ * session_set_save_handler($handler, true);
+ * session_save_path('sessions');
+ * session_start();
+ *
+ * // Then write and read the $_SESSION array.
+ * $_SESSION['name'] = 'Bob';
+ * echo $_SESSION['name'];
+ * ```
+ *
+ * ```
+ * // Session handler with error handling:
+ * use Google\Cloud\Firestore\FirestoreClient;
+ *
+ * $firestore = new FirestoreClient();
+ *
+ * $handler = $firestore->sessionHandler();
+ * session_set_save_handler($handler, true);
+ * session_save_path('sessions');
+ * session_start();
+ *
+ * // Then read and write the $_SESSION array.
+ * $_SESSION['name'] = 'Bob';
+ *
+ * function handle_session_error($errNo, $errStr, $errFile, $errLine) {
+ *     // We throw an exception here, but you can do whatever you need.
+ *     throw new RuntimeException(
+ *         "$errStr in $errFile on line $errLine",
+ *         $errNo
+ *     );
+ * }
+ * set_error_handler('handle_session_error', E_USER_WARNING);
+ *
+ * // If `write` fails for any reason, an exception will be thrown.
+ * session_write_close();
+ * restore_error_handler();
+ *
+ * // You can still read the $_SESSION array after closing the session.
+ * echo $_SESSION['name'];
+ * ```
+ *
+ * @see http://php.net/manual/en/class.sessionhandlerinterface.php SessionHandlerInterface
+ */
+class FirestoreSessionHandler implements SessionHandlerInterface
+{
+    use SnapshotTrait;
+
+    /**
+     * @var ConnectionInterface
+     */
+    private $connection;
+    /**
+     * @var ValueMapper
+     */
+    private $valueMapper;
+    /**
+     * @var string
+     */
+    private $projectId;
+    /**
+     * @var string
+     */
+    private $database;
+    /**
+     * @var array
+     */
+    private $options;
+    /**
+     * @var string
+     */
+    private $savePath;
+    /**
+     * @var string
+     */
+    private $sessionName;
+    /**
+     * @var Transaction
+     */
+    private $transaction;
+
+    /**
+     * Create a custom session handler backed by Cloud Firestore.
+     *
+     * @param ConnectionInterface $connection A Connection to Cloud Firestore.
+     * @param ValueMapper $valueMapper A Firestore Value Mapper.
+     * @param string $projectId The current project id.
+     * @param string $database The database id.
+     * @param array $options [optional]
```

Comment:
Can you please copy the options from `FirestoreClient::sessionHandler()` to this docblock?